### PR TITLE
Catalog of my things

### PR DESCRIPTION
**Pull request by MAleemH**
### PR Includes:

- [x] `Book` class, `Label` class, `MusicAlbum` class, `Genre` class, `Game` class, `Author` class, `Movie` class, and `Source` class in separate .rb files.
- [x] `add_item` method that takes an instance of the `Item` class as an input and adds the input item to the collection of items.
- [x] `can_be_archived?()` method.
- [x] Unit tests for all implemented methods.
- [x] `main.rb` file that will serve as console app entry-point.
- [x] All data can be preserved by saving collections in .json files.
- [x] A `schema.sql` file with tables that will be analogical to the structure of the classes in the app.
- [x] [Video](https://drive.google.com/file/d/1cPW0wj0K0mO6XAC7neJz6yidUPr5zZCy/view?usp=sharing) presentation.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+    puts welcome
+    loop do
+      list_options
+      choice = gets.chomp
+
+      case choice
+      when '1'
+        list_books
+      when '2'
+        display_music_album(show_index: true)
```
Comment (elisha2kyakpo1) [inline app.rb:174]: - [ ] Kindly fix the error that shows when a user click option 2 and 5. I have added an image for your reference.

error

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+  #     @music_albums.each_with_index do |music_album, index|
+  #       puts "#{index}) Publish Date: #{music_album['publish_date']} On Spotify: #{music_album['on_spotify']}, ID: #{music_album['id']}"
+  #     end
+  #     puts 'Display music album with index successfully'
+  #   else
+  #     @music_albums.each { |music_album| puts "Publish Date: #{music_album['publish_date']} On Spotify: #{music_album['on_spotify']}, ID: #{music_album['id']}" }
+  #     puts 'Display music album without index successfully'
+  #   end
+  #   run
+  # end
```
Comment (elisha2kyakpo1) [inline app.rb:305]: - [ ] Kindly remove all the commented code to make the codebase look clean.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description are good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.
✔️ Tests passing👏

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly work on all the JSON data files, currently, data entered by users is not being saved, The requirement for data storage is that, when a user closes the app and opens it again, the data they entered before closing should be available. I have not commented on individual files because the issue is for all the JSON files. (In the video, I saw data being saved in JSON files, make sure you have all the code in the dev branch which is the base (GitHubflow).
- [x] Let's record another video because you did not follow some of the Project documentation (video presentation). I have attached an image with a red mark on the requirements you did not cover in the video.

video


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @lovelyteam😉, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] The app work as expected ✔️ 
- [ ] All tests are passing ✔️ 
- [ ] Correct Gitflow and GitHub flow ✔️ 
- [ ] Good commit history ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**